### PR TITLE
Week 2 and 3

### PR DESCRIPTION
**Pull request by Shweta251093**
Please review Week 2 & 3 code

### CODE REVIEW

File: src/Shweta/Week2/TestPassByValue.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package Shweta.Week2;
+
+import java.util.Arrays;
+public class TestPassByValue {
+    public static void main(String[] args) {
+        int[] nums = {1,2,3,4,5};
+        System.out.println(Arrays.toString(nums)); //=> [1,2,3,4,5]
+        magic(nums);
+        System.out.println(Arrays.toString(nums)); //=> [1,2,3,4,5]
```
Comment (kushalxp) [inline src/Shweta/Week2/TestPassByValue.java:23]: How does the output looks like ? is it => [1,2,3,4,5] ? & what is your finding ?

File: src/Shweta/Week3/PlusOne.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                }else {
+                    digits[i] = digits[i] + 1;
+                    return digits;
+                }
+            }
+        }
+        if(carry==1){
+            plusOne[0] = 1;
+            for(int i=1; i<=digits.length;i++){
+                plusOne[i] = digits[i-1];
```
Comment (kushalxp) [inline src/Shweta/Week3/PlusOne.java]: DO we need to copy remaining digit ? what happens when carrry is one on line 29 ? how does other elements change when carry is 1 ?

File: src/Shweta/Week2/Pivot.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            rightSum = rightSum - nums[i] ;
+            if(rightSum==leftSum){
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    // nums = [-1,-1,0,1,1,0] => 5
+    public static int optimizePivot_Bug(int[] nums){
```
Comment (kushalxp) [inline src/Shweta/Week2/Pivot.java:65]: although this code works ? does it make sense on your variable name use ?

**Review by kushalxp — CHANGES_REQUESTED** [review]
Have a look at comments and let me know your thoughts

File: src/Shweta/Week2/TestPassByValue.java (hunk 1 above)
Comment (Shweta251093) [reply src/Shweta/Week2/TestPassByValue.java:23]: Output is [5,1,2,3,4] not  [1,2,3,4,5]
As original array is changed it means reference is used when array was passed.

File: src/Shweta/Week2/Pivot.java (hunk 3 above)
Comment (Shweta251093) [reply src/Shweta/Week2/Pivot.java:65]: yes, my right was actually left and left was right..until realised😅

**Review by kushalxp — APPROVED** [review]
LGTM

**Pull request merged**